### PR TITLE
Add option to disable repair in database

### PR DESCRIPTION
**Pull request by cw9**
For the m3db in collector, there will be no peers to repair from,
should have an option to just turn repair off and save some resource

### CODE REVIEW

**Comment by cw9** [comment]
Did we use to have a toggle like this and then removed? 

@xichen2020 @robskillington @prateek

**Comment by xichen2020** [comment]
Enable/Disable repair is done via namespace-level options

**Comment by cw9** [comment]
Then for cases like this (don't need repair), I still need to create an admin client because the database has to create a repairer and that requires the repair options to have an admin client to be valid.

**Comment by xichen2020** [comment]
In that case I suggest
- adding a new noopRepairer
- adding a `NewDatabaseRepairerFn` to the database options
- in your usecase, make the repairer ctor return the noopRepairer

**Comment by cw9** [comment]
I added the no-op repairer, I'm hesitating on exposing the NewDatabaseRepairerFn() in the Options, because for the use case we have, we basically just need to toggle between 2 repairers.

**Comment by xichen2020** [comment]
hm why not? This makes it very flexible in terms of what type of repairer the database uses, even though for our usecases it's either a no-op repairer or a full-fledged repairer, but it doesn't have to be and other users of the repo can pass in their custom repairer if they choose to

**Comment by cw9** [comment]
because the way we create the databaseRepairer right now requires both database interface and databaseRepairer interface. It seems to be easy to expose the databaseRepairer interface, but I'm not sure about changing database to Database

File: storage/repair.go [hunk 1]
```diff
@@ -606,3 +606,13 @@ func (r *dbRepairer) repairWithTimeRange(tr xtime.Range) error {
...
 }
+
+type noopRepairer struct{}
+
+func newNoopDatabaseRepairer() databaseRepairer {
+	return noopRepairer{}
+}
+
+func (r noopRepairer) ShouldRun(start time.Time) bool    { return false }
+func (r noopRepairer) Run(start time.Time, mode runType) { return }
```
Comment (xichen2020) [inline storage/repair.go]: nit: `{}` is slightly cleaner

**Pull request merged**